x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2855 @@
+  // [...]
+
+  immutable: IsRuleImmutable,
+  ruleSource: RuleSource.transform(camelize).optional(),
+  // [...]
+});
+```
+
+> Notice that in the internal schema we cannot simply reuse the `RuleSource` attribute defined for the API schema, since it should have CamelCase and the API schema uses snake_case. We need to apply a transformation to our Zod type. See this [issue](https://github.com/colinhacks/zod/issues/486#issuecomment-1567296747).
+
+In the internal rule schema, there are two additional important reasons why we need to make sure that these two fields optional:
+
+- When rules are executed, a call to the method `validateRuleTypeParams` is done, which is a method that validates the passed rule's parameters using the validators defined in `x-pack/plugins/security_solution/server/lib/detection_engine/rule_types`, within each of the rule query types files (for [EQL rules](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/eql/create_eql_alert_type.ts#L27), for example). The validation is done based on the internal rule schema `BaseRulesParams` displayed above. Having `ruleSource` as required field would cause custom rules or prebuilt rules that haven't had their schema migrated to fail during runtime.
+- The Rule Client `update` method also calls the `validateRuleTypeParams` to validate the rule's params. Since the Rule Client's `update` method is used in our endpoint handlers, such as and `/rules/patch` and `/_bulk_actions`, these would fail when executed against a payload of custom rule.
+
+##### Prebuilt Rule asset schema
+
+The `PrebuiltRuleAsset` type needs to be updated to include the new `source_updated_at` date that will be progressively shipped with new versions of rules in the Elastic Prebuilt Rules package.
+
+Notice that this field will be a **top-level field in the Prebuilt Rule asset** schema, but will be part of the `rule_source` field in the API rule schema and internal rule schema.
+
+_Source: [x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/model/rule_assets/prebuilt_rule_asset.ts](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/model/rule_assets/prebuilt_rule_asset.ts)_
+
+```ts
+export const PrebuiltRuleAsset = BaseCreateProps.and(TypeSpecificCreateProps).and(
+  z.object({
+    rule_id: RuleSignatureId,
+    version: RuleVersion,
+    related_integrations: RelatedIntegrationArray.optional(),
+    required_fields: RequiredFieldArray.optional(),
+    setup: SetupGuide.optional(),
+    source_updated_at: SourceUpdatedAt.optional(), // new optional field
+  })
+);
+```
+
+### Deprecating the `immutable` field
+
+In order to mark the `immutable` field as deprecated, and making sure that our application and API users are aware that the field has been deprecated and replaced by the `rule_source` field, we will communicate this change in three ways:
```

Comment:
No. As we discussed in the last Tech Time meeting, since we are not introducing any breaking changes to the API we won't be using API versioning. This avoids the complexity of maintaing two different versions, with the new one being backwards compatible with the older one.